File: C programming/Chapter-2/Problem6.c
/*      PROBLEM 6
 * Two numbers are input through two locations c and d.
 * Write a program to interchange contents of c and d.
 */

// include libraries
#include <stdio.h>

int main()
{
    // declare variables
    int a, b;

    // take the input
    printf("Enter the integers: \n");
    scanf("%d %d", &a, &b);

    // print the variables before swapping
    printf("Before Swapping: \n");
    printf("The first number: %d\n", a);
    printf("The second number: %d\n", b);

    // swap the variables using a third variable
    int temp;
    temp = a;
    a = b;
    b = temp;

    // print the variables after swapping
    printf("After Swapping: \n");
    printf("The first number: %d\n", a);
    printf("The second number: %d\n", b);
}

File: Chapter-2/Java/Problem6.java

//import required libraries
import java.util.Scanner;

public class Problem6 {
    public static void main(String[] args) {
        // declare variables
        int a, b;
        int temp = 0;
        // take input
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter two numbers: ");
        a = sc.nextInt();
        b = sc.nextInt();
        System.out.printf("Before Swap: a = %d, b = %d\n", a, b);
        // swap
        temp = a;
        a = b;
        b = temp;

        // print
        System.out.printf("After Swap: a = %d, b = %d", a, b);
        sc.close();
    }
}
